.github/workflows/get-previous-release/action.yml

```diff
@@ -0,0 +1,42 @@
+name: Get Previous Release
+description: Get the previous release name from the current tag and set it as an environment variable.
+
+inputs:
+  CHANNEL:
+    required: true
+    description: "The channel to get the version to (ga, cloud, beta)."
+
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        #if channel is cloud or beta
+
+        if [[ ${{ inputs.CHANNEL }} == "cloud" || ${{ inputs.CHANNEL }} == "beta" ]]; then
+           tags=$(git ls-remote --tags | grep ${{ inputs.CHANNEL }} | grep -v -E '.*-rc.*' | awk '{print $2}')
+        fi
+
+        if [[ ${{ inputs.CHANNEL }} == "ga" ]]; then
+           tags=$(git ls-remote --tags | grep "refs/tags/v[0-9]\+\.[0-9]\+\.[0-9]\+$")
+        fi
+
+        git fetch --tags
```

Comment:
Why do we need this?

Reply:
This action was built to get the last release of a specific channel.
This parameter `PREVIOUS_TAG_SHA` will be used in the next action `./.github/workflows/generate-changelog` as `BASE_TAG_NAME` while the `HEAD_BRANCH_NAME` is the current `${{ github.ref }}`

Reply:
Added.

Reply:
Move this to after the comment ("loop through the list...")